**Теория вероятностей и математическая статистика**


**Урок 5.** Проверка статистических гипотез. P-значения. Доверительные интервалы. A/B-тестирование

Домашнее задание  
автор: **Игорь Бертяев**

### задание 1

Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16.  
Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95,  
если выборочная средняя M = 80, а объем выборки n = 256.

Решение:  
используя формулу доверительного интервала

$$T_{1,2} = \overline{X} \pm Kp \frac{\sigma}{\sqrt{n}}$$ 
  
подставив наши значения (Kp для 2,5% = 1,96):

$$T_{1,2} = 80 \pm 1.96 \frac{16}{\sqrt{256}}$$ 

In [2]:
import numpy as np

In [6]:
T1 = 80 + 1.96*(16/np.sqrt(256))
T1

81.96

In [7]:
T2 = 80 - 1.96*(16/np.sqrt(256))
T2

78.04

итого доверительный интервал получается [78.04;81.96] 

### задание 2

В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью,  
получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1   
Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей,  
оценить истинное значение величины X при помощи доверительного интервала,  
покрывающего это значение с доверительной вероятностью 0,95. 


Решение:  
также используем формулу доверительного интервала

$$T_{1,2} = \overline{X} \pm Kp \frac{\sigma}{\sqrt{n}}$$ 
  
критерий для 95% будет равен Kp = 1,96 (для каждой стороны)  
среднее значение выборки будет:

In [12]:
x = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
x.mean()

6.590000000000001

также нам нужно найти сигму (СКО)

In [15]:
x.std()

0.42766809560686203

теперь подставляем значения в формулу:

In [17]:
T_1 = x.mean() + 1.96*(x.std()/np.sqrt(10))
T_1

6.855071431882051

In [20]:
T_2 = x.mean() - 1.96*(x.std()/np.sqrt(10))
T_2

6.324928568117951

таким образом мы нашли что величина Х с вероятностью 95% должна находится в интервале [6.325;6.855]

### задание 3 

Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.  
Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков  
средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 кв.мм. 

Решение:  
H0 = 17  
H1 = 17.5  
сравним критерий табличный и наблюдаемый, используем Z-тест

In [29]:
Zn = (17.5-17) / (np.sqrt(4)/np.sqrt(100))
Zn

2.5

Теперь, зная табличную величину Zt = 1.645 и наблюдаемую Zn = 2.5 можем заключить что Н0 не верна при а=0,05, т.е. исходную гипотезу мы отвергаем. 

### задание 4

Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек.  
Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190. 
Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, 
что доверительная вероятность равна 99%?

Решение: не совсем понял как быть с двусторонним тестом (та же формула или нет) поэтому я решаю как односторонний.  
так же как и в предыдущей задаче сравним критерии - табличный и наблюдаемый.  
табличный критерий найдем исходя из того что 1%  по таблице найдем Z = 2.33

In [27]:
# найдем среднее
z = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])  
z.mean()  

198.5

In [30]:
z.std()

4.224926034855522

In [31]:
Z_n = (198.5-200) / (np.sqrt(4.225)/np.sqrt(10))
Z_n

-2.307692307692308

в результате -2,308 полуучается лежит внутри основной площади - т.е. ближе табличного значения -2,33, значит гипотиза продавца верная.  

Хотя на ппервый взгляд получается что М0 не равна М, т.е. уже не двусторонняя, немнго тут не понятно выходит. 